recipe/meta.yaml

```diff
@@ -164,8 +164,7 @@ outputs:
         - suitesparse
         - zlib
         - fenics-ffc =={{ version }}
-        - hdf5 * {{ mpi_prefix }}_*
-        - {{ pin_compatible("boost-cpp", exact=True) }}
```

Comment:
`{{ pin_compatible(..., exact=True) }}` means that it pins to the exact build (i.e., on the build string level).
Here you rather want a plain `boost-cpp` in `host`/`run` so that `conda-build` will replace it with the pinnings definition.

Reply:
The `fenics` package requires the exact `boost-cpp` version which is the cause of https://github.com/conda-forge/mshr-feedstock/issues/18

Reply:
> exact `boost-cpp` version 

Exact *version* would be understandable. But the current `conda-forge-pinning` already has a `max_pin: x.x.x` for `boost-cpp` which means if you have it in `host` and `run`, then you'd get a pin to `>=1.72.0,<1.73.0a0`.
Exact *build* is what doesn't seem sensible to me.

Reply:
If you don't trust `conda-forge-pinning` (e.g., to avoid a `max_pin` change affecting this recipe), you can do `{{ pin_compatible("boost-cpp", max_pin="x.x.x") }}` yourself.